### PR TITLE
Fix markT, unmarkT and deleteT bugs

### LINKED ISSUE #161: Output message for markT, unmarkT and deleteT could be improved
Previously, the descriptive exceptions thrown by `ParserUtil::parseIndex` and `ParserUtil::parseIndexes` were accidentally caught (and a more general error message was thrown) instead of being displayed to the user as error messages.

### PR DESCRIPTION
**Pull request by Chandan8186**
Fix #161, #162

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/DeleteTaskCommandParser.java [hunk 1]
```diff
@@ -34,18 +34,13 @@ public DeleteTaskCommand parse(String args) throws ParseException {
...
-
-            return new DeleteTaskCommand(projectIndex, taskIndexes);
-        } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTaskCommand.MESSAGE_USAGE), pe);
-        }
+        // get project and task indexes
+        Index projectIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_PROJECT).get());
+        List<Index> taskIndexes = ParserUtil.parseIndexes(argMultimap.getValue(PREFIX_TASK).get());
+        assert taskIndexes.size() > 0;
```
Comment (antonTan96) [inline src/main/java/seedu/address/logic/parser/DeleteTaskCommandParser.java:40]: Good job on identifying that the source of the problem is the try-catch block!

**Review by antonTan96 — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/parser/MarkTaskCommandParser.java [hunk 2]
```diff
@@ -33,17 +33,12 @@ public MarkTaskCommand parse(String args) throws ParseException {
...
-        } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkTaskCommand.MESSAGE_USAGE), pe);
-        }
+        // get project and task indexes
+        Index projectIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_PROJECT).get());
+        List<Index> taskIndexes = ParserUtil.parseIndexes(argMultimap.getValue(PREFIX_TASK).get());
+        assert taskIndexes.size() > 0;
+
+        return new MarkTaskCommand(projectIndex, taskIndexes);
```
Comment (aslam341) [inline src/main/java/seedu/address/logic/parser/MarkTaskCommandParser.java:41]: I believe this solves the "duplicate index error message not showing up" issue right?

**Review by aslam341 — APPROVED** [review]
LGTM.

File: src/main/java/seedu/address/logic/parser/MarkTaskCommandParser.java (hunk 2 above)
Comment (Chandan8186) [reply src/main/java/seedu/address/logic/parser/MarkTaskCommandParser.java:41]: It's not this part, it's the removal of the try-catch block from the `parse` function (this block unintentionally caught some exceptions).

**Pull request merged**